tests/test_cli.py

```diff
@@ -11,36 +13,36 @@ def test_cli_options_from_env(caplog, runner):
     assert "Starting credit card slips process" in caplog.text
     assert (
         "Credit card slips processing complete for date 2023-01-02. Email sent to "
-        "recipient(s) ('recipient1@example.com', 'recipient2@example.com')"
-        in caplog.text
+        "recipient(s) ('recipient1@example.com', 'recipient2@example.com')" in caplog.text
     )
 
 
 def test_cli_all_options_passed(caplog, runner):
-    result = runner.invoke(
-        main,
-        [
-            "--source-email",
-            "from@example.com",
-            "--recipient-email",
-            "recipient1@example.com",
-            "--recipient-email",
-            "recipient2@example.com",
-            "--date",
-            "2023-01-02",
-            "--log-level",
-            "debug",
-        ],
-    )
-    assert result.exit_code == 0
-    assert "Logger 'root' configured with level=DEBUG" in caplog.text
-    assert (
-        "Command called with options: {'source_email': 'from@example.com', "
-        "'recipient_email': ('recipient1@example.com', 'recipient2@example.com'), "
-        "'date': '2023-01-02', 'log_level': 'debug'}" in caplog.text
-    )
-    assert (
-        "Credit card slips processing complete for date 2023-01-02. Email sent to "
-        "recipient(s) ('recipient1@example.com', 'recipient2@example.com')"
-        in caplog.text
-    )
+    with caplog.at_level(logging.DEBUG):
```

Comment:
This definitely works, but I think we can also do `caplog.set_level("DEBUG")` as an option as well.  I did this in geo-harvester to avoid shifting the entire test into a context manager. 

Think they both have their place, and this totally works!   Just noting.

Reply:
I like this better, fixed!